stores/UserProfileStore.js

```diff
@@ -198,6 +218,11 @@ class UserProfileStore extends BaseStore {
         this.emitChange();
     }
 
+    handleSocialSignInError(err) {
+        this.socialLoginError = true;
```

Comment:
You might be able to reset it immediately after `this.emitChange()` instead of having a full component --> action --> store cycle again.